home-page-script.js

```diff
@@ -0,0 +1,194 @@
+    picture_url:
+      "https://meyers.dk/media/13095/roedgroed_med_floede.jpg?anchor=center&mode=crop&width=1260&format=jpeg&quality=80&rnd=132169219430000000",
+    ingredients: [
+      { name: "Strawberries", amount: "300 g" },
+      { name: "Red currants", amount: "200 g" },
+      { name: "Sugar", amount: "100 g" },
+      { name: "Water", amount: "250 ml" },
+      { name: "Cornstarch", amount: "2 tbsp" },
+      { name: "Heavy cream", amount: "100 ml" },
+    ],
+    description:
+      "Cook berries with sugar, thicken with cornstarch, and serve with cream.",
+  },
+  {
+    id: 5,
+    title: "Frikadeller",
+    picture_url:
+      "https://ff234d58.rocketcdn.me/wp-content/uploads/2020/10/IMG_1100-860x802.jpg",
+    ingredients: [
+      { name: "Ground pork", amount: "500 g" },
+      { name: "Egg", amount: "1" },
+      { name: "Milk", amount: "100 ml" },
+      { name: "Onion", amount: "1" },
+      { name: "Flour", amount: "2 tbsp" },
+      { name: "Salt", amount: "1 tsp" },
+      { name: "Pepper", amount: "0.5 tsp" },
+      { name: "Butter", amount: "50 g" },
+    ],
+    description:
+      "Mix the ingredients, form small patties, and fry until golden brown.",
+  },
+];
+
+recipeObjectArray.forEach((dish) => {
+  dish.ingredients.sort((a, b) => {
+    const numberA = parseFloat(a.amount) || 0;
+    const numberB = parseFloat(b.amount) || 0;
+    return numberA - numberB;
+  });
+});
+
+document.querySelector("#create-page").addEventListener("click", function () {
+  window.location.href = "recipe-form.html";
+});
+
+//<div class="recipe-box">
+//<h1 class="recipe-name">Recipe Name</h1>
+//<img class="recipe-img" src="" alt="Recipe Image" >
+/*<h1 class="ingredients">Ingredients:</h1>
+<ul class="ingredient-list"></ul>
+<h1 class="description">Description:</h1>
+<p class="recipe-description"></p>
+</div> */
+const recipeContainer = document.querySelector(".recipe-container");
+
+function displayRecipe(recipe) {
+  recipe.forEach((dish) => {
+    const recipeBox = document.createElement("div");
+    recipeBox.classList.add("recipe-box");
+    const recipeName = document.createElement("h1");
+    recipeName.classList.add("recipe-name");
+    recipeName.innerHTML = dish.title;
+    const recipeImg = document.createElement("img");
+    recipeImg.classList.add("recipe-img");
+    recipeImg.src = dish.picture_url;
+    recipeImg.alt = "Recipe Image";
+    const recipeIngredients = document.createElement("h1");
+    recipeIngredients.innerHTML = "Ingredients";
+    recipeIngredients.classList.add("ingredients");
+    const ingredientList = document.createElement("ul");
+    ingredientList.classList.add("ingredient-list");
+    dish.ingredients.forEach((ingredient) => {
+      const li = document.createElement("li");
+      if (typeof ingredient.amount !== "string") {
+        li.innerHTML = `${ingredient.name}: not provided`;
+      } else {
+        li.innerHTML = `${ingredient.name}: ${ingredient.amount}`;
+      }
```

Comment:
If `ingredient.amount` is not provided by default you will get `undefined` which wouldn't be very different than `not provided`. I think you can safely remove this check, and replace it with:
```suggestion
li.innerHTML = `${ingredient.name}: ${ingredient.amount}`;
```